pkg/cmd/experimental/policy/policy.go

```diff
@@ -79,23 +81,36 @@ func getUniqueName(basename string, existingNames *util.StringSet) string {
 	return string(util.NewUUID())
 }
 
-func getExistingRoleBindingsForRole(roleNamespace, role, bindingNamespace string, client *client.Client) ([]*authorizationapi.RoleBinding, *util.StringSet, error) {
+func getExistingRoleBindingsForRole(roleNamespace, role, bindingNamespace string, client *client.Client) ([]*authorizationapi.RoleBinding, error) {
 	existingBindings, err := client.PolicyBindings(bindingNamespace).Get(roleNamespace)
 	if err != nil && !strings.Contains(err.Error(), " not found") {
-		return nil, &util.StringSet{}, err
+		return nil, err
 	}
 
 	ret := make([]*authorizationapi.RoleBinding, 0)
-	roleBindingNames := &util.StringSet{}
 	// see if we can find an existing binding that points to the role in question.
 	for _, currBinding := range existingBindings.RoleBindings {
-		roleBindingNames.Insert(currBinding.Name)
-
 		if currBinding.RoleRef.Name == role {
 			t := currBinding
 			ret = append(ret, &t)
 		}
 	}
 
-	return ret, roleBindingNames, nil
+	return ret, nil
+}
+
+func getExistingRoleBindingNames(bindingNamespace string, client *client.Client) (*util.StringSet, error) {
```

Comment:
I don't know how pervasive this is in your code, but you shouldn't be passing around full client objects.  You could have passed a PolicyBindingInterface here instead of passing namespace and client.